### PR TITLE
Add Python 3.12 support

### LINKED ISSUE #324: Support Python 3.12
### Problem Description
Python 3.12 was released on Oct 2, 2023.

### Expected behavior
In order to upgrade CTGAN, we'll need to:
- Ensure that our external dependencies are compatible with 3.12 (eg. pytorch, numpy, etc.)
- Upgrade the internal libraries in the SDV ecosystem to 3.12 (RDT). We will file separate issues for these in each of the libraries.

This is a blocking issue for upgrading the main SDV library to 3.12. See [SDV Issue 1704](https://github.com/sdv-dev/SDV/issues/1704)

### PR DESCRIPTION
**Pull request by fealho**
CU-86ayu7163, Resolve #324.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86ayu7163 CTGAN - Support Python 3.12 #324](https://app.clickup.com/t/86ayu7163)

**Review by pvk-developer — APPROVED** [review]

File: tasks.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def _get_minimum_versions(dependencies, python_version):
     for dependency in dependencies:
         if '@' in dependency:
             name, url = dependency.split(' @ ')
-            min_versions[name] = f'{name} @ {url}'
+            min_versions[name] = f'{url}#egg={name}'
```
Comment (amontanez24) [inline tasks.py:60]: @R-Palazzo Is there anything else that should be done here?

Comment (R-Palazzo) [reply tasks.py:60]: I also updated the test here:
https://github.com/sdv-dev/CTGAN/blob/0d4708a3b0e64540785d776dc17285966eed477d/tests/test_tasks.py#L18

In this PR:
https://github.com/sdv-dev/CTGAN/pull/356

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py39-lint, py3{8,9,10,11}-{unit,integration,readme}
+envlist = py39-lint, py3{8,9,10,11,12}-{unit,integration,readme}
 
 [testenv]
 skipsdist = false
```
[the remark this answers is not in the document]
Comment (fealho) [reply tox.ini:5]: The makefile commands `clean-test` and `test-all` use tox.

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**